File: shop-admin-ui/src/api/modules/logistics.ts
import { get, post, put } from '../request'

export interface Logistics {
  id: number
  orderId: number
  logisticsNo: string
  company: string
  companyCode: string
  status: number
  senderName: string
  senderPhone: string
  senderAddress: string
  senderLongitude?: number
  senderLatitude?: number
  receiverName: string
  receiverPhone: string
  receiverAddress: string
  receiverLongitude?: number
  receiverLatitude?: number
  lastQueryTime: string
  createTime: string
}

export interface LogisticsTrace {
  id: number
  logisticsId: number
  traceTime: string
  content: string
  location: string
  latitude: number
  longitude: number
  createTime: string
}

export interface CreateLogisticsRequest {
  orderId: number
  logisticsNo: string
  company: string
  senderName?: string
  senderPhone?: string
  senderAddress?: string
  senderLongitude?: number
  senderLatitude?: number
  receiverName?: string
  receiverPhone?: string
  receiverAddress?: string
  receiverLongitude?: number
  receiverLatitude?: number
}

// 物流公司列表
export const logisticsCompanies = [
  { name: '顺丰速运', code: 'shunfeng' },
  { name: '韵达快递', code: 'yunda' },
  { name: '圆通速递', code: 'yuantong' },
  { name: '中通快递', code: 'zhongtong' },
  { name: '申通快递', code: 'shentong' },
  { name: '京东物流', code: 'jd' },
  { name: 'EMS', code: 'ems' },
  { name: '德邦快递', code: 'debang' },
  { name: '极兔速递', code: 'jtexpress' },
  { name: '百世快递', code: 'huitongkuaidi' }
]

// 创建物流
export const createLogistics = (data: CreateLogisticsRequest) => {
  return post<Logistics>('/api/v1/logistics', data)
}

// 查询订单物流
export const getLogisticsByOrderId = (orderId: number) => {
  return get<Logistics>(`/api/v1/logistics/order/${orderId}`)
}

// 获取物流详情
export const getLogisticsDetail = (id: number) => {
  return get<Logistics>(`/api/v1/logistics/${id}`)
}

// 获取物流轨迹
export const getLogisticsTraces = (id: number) => {
  return get<LogisticsTrace[]>(`/api/v1/logistics/${id}/traces`)
}

// 刷新物流
export const refreshLogistics = (id: number) => {
  return post(`/api/v1/logistics/${id}/refresh`)
}

File: shop-backend/shop-modules/shop-order-service/src/main/java/com/shop/order/entity/Logistics.java
package com.shop.order.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;

/**
 * 物流信息实体
 *
 * @author shop
 * @since 2026-06-07
 */
@Data
@TableName("oms_logistics")
public class Logistics implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    /** 订单ID */
    private Long orderId;

    /** 物流单号 */
    private String logisticsNo;

    /** 物流公司 */
    private String company;

    /** 状态: 0-已发货 1-运输中 2-派送中 3-已签收 */
    private Integer status;

    /** 发件人姓名 */
    private String senderName;

    /** 发件人电话 */
    private String senderPhone;

    /** 发件人地址 */
    private String senderAddress;

    /** 发件人纬度 */
    private BigDecimal senderLatitude;

    /** 发件人经度 */
    private BigDecimal senderLongitude;

    /** 收件人姓名 */
    private String receiverName;

    /** 收件人电话 */
    private String receiverPhone;

    /** 收件人地址 */
    private String receiverAddress;

    /** 收件人纬度 */
    private BigDecimal receiverLatitude;

    /** 收件人经度 */
    private BigDecimal receiverLongitude;

    /** 上次查询API时间 */
    private LocalDateTime lastQueryTime;

    @TableField(exist = false)
    private List<LogisticsTrace> traces;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createTime;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updateTime;

    @TableLogic
    @TableField(fill = FieldFill.INSERT)
    private Integer deleted;
}
